### DISCUSSION TITLE
Mehrere URL Parameter bei shared maps ziehen nicht

### DISCUSSION THREAD

**Discussion by ue-black** [e-netze/webgis-community#133] [General]
Hallo!

Unser Straßenbau möchte aus einer externen Anwendung heraus den Atlas starten. Die übergebenen Parameter sind KG-Nummer und Grundstück. Sie haben eine eigene Karte gebaut, weil ihnen unser Design nicht ganz gepasst hat:

https://atlas.vorarlberg.at/portal/map/_sharedmaps/0xA582EFFFFF1C542C5C5B2319D2B73736D7FB58823F3A500816AAADBD0CCD459B?abfragethema=grundstueck&kgnr=91107&grundstueck=168/2

Die Abfrage berücksichtigt nur die KG-Nummer. Hier ein Video wo man sieht, dass zuerst der erste und dann der zweite Parameter verwendet wird:

https://github.com/gis-eni/webgis-forum/assets/127403099/5abe0a6e-4438-4880-89c6-e6e28725ccb5

Bei einer Standardkarte funktioniert der exakt gleiche Aufruf mit mehreren Werten:

https://atlas.vorarlberg.at/portal/map/Basiskarten/Kataster%20-%20DKM?abfragethema=grundstueck&kgnr=91107&grundstueck=168/2

**Comment by gis-eni** [comment]
Was ich nicht ganz verstehe:

Ich habe in der Oberfläche keine Möglichkeit die KG einzugeben:

<img width="238" alt="image" src="https://github.com/gis-eni/webgis-forum/assets/62870568/5ef7c05b-4b32-4995-a95c-ed678818c5e1">


Wenn ich diese Url Eingebe funktioniert es:



https://atlas.vorarlberg.at/portal/map/_sharedmaps/0xA582EFFFFF1C542C5C5B2319D2B73736D7FB58823F3A500816AAADBD0CCD459B?abfragethema=grundstueck&gemeinde=Bregenz&grundstueck=168/2

Also dürfte es nicht an den mehreren Parametern liegen.

Und: Die Gemeinde ist Case Sensitiv... Absicht?

**Reply by ue-black** [reply]
Der Hinweis mit "nicht sichtbar" ist gut. Daran wird es liegen. Wir haben bei der Detailsuche nur Gemeinde und Grundstücksnummer sichtbar geschaltet. Die KG-Nummer verwirrt die meisten User, deshalb haben wir den Haken entfernt.
![image](https://github.com/gis-eni/webgis-forum/assets/127403099/15ad246a-1149-4c97-81e6-6369fc67766a)

Im Testsystem haben wir die KGNr jetzt testweise ergänzt. Jetzt funktioniert die Abfrage auch bei einer shared map im Testsystem.
![image](https://github.com/gis-eni/webgis-forum/assets/127403099/e9be21b2-602b-4c4f-8020-792fcef33bad)

Der Parameter Gemeinde ist nicht verwendbar, es gibt dann keinen eindeutigen Treffer. Gemeinde zu Katastralgemeinde = 1:n 
z.B. hat Feldkirch mehrere KGs: 
https://atlas.vorarlberg.at/portal/map/Basiskarten/Kataster%20-%20DKM?abfragethema=grundstueck&gemeinde=Feldkirch&grundstueck=.1

Das erklärt aber noch immer nicht warum bei shared maps im Vergleich zu vorkonfigurierten Karten nicht sichtbare Abfragen nicht funktionieren.

**Reply by gis-eni** [reply]
Das die Abfrage mit Gemeinde und GNR nicht eindeutig ist, ist mir klar. Drum wundert es mich ja, warum ihr das so anbietet. Bei uns ist es genau umgekehrt. Wenn jemand eine Grundstücksnummer hat, hat er in der Regel auch die KG Nummer (oder den KG Namen) dazu. Und das ist dann eindeutig. Aber kennt euerer Anwender natürlich besser als ich ;)

**Comment by gis-eni** [comment]
Und: die Abfrage Id ist bei dir nicht eindeutig.

``grunstueck`` gibt es bei dir sowohl in **Orientierung** als auch in **DKM Stichtagsdaten**.
Einmal ist der Dient unten und einmal ganz oben eingebunden... Die Konvention bei den Url Parametern ist (kommt noch von Webgis 2), dass man die Abfrage Id übergibt. Eine davon musst du anders benennen oder raus schmeißen.

**Reply by ue-black** [reply]
Es könnte auch an der doppelten ID liegen, stimmt. Wir werden das morgen überprüfen. Im Testsystem hat es komischerweise auch mit doppelter ID funktioniert. Wir werden das morgen produktiv nochmals testen indem wir die ID für `grundstueck` in **Orientierung** ändern.

**Reply by gis-eni** [reply]
Es sind definitiv die doppelten Abfrage Ids. Warums im Testsystem funktioniert, kann ich nicht sagen. Vielleiht nimmer er da zufällig die richtige, oder die Reihenfolge der Dienste ist anders. Kann auch sein, dass er beide abfragt und der letzte der das Ergebnis liefert gewinnt. So schaut es bei der Shared Map auch aus...
